srtcore/srt_c_api.cpp

```diff
@@ -314,4 +314,16 @@ int srt_listen_callback(SRTSOCKET lsn, srt_listen_callback_fn* hook, void* opaq)
     return CUDT::installAcceptHook(lsn, hook, opaq);
 }
 
+int srt_getversion(void* val, int len)
+{
+    if (!val)
+        return CUDT::setError(CUDTException(MJ_NOTSUP, MN_INVAL));
+
+    if (len != sizeof(uint32_t))
+        return CUDT::setError(CUDTException(MJ_NOTSUP, MN_INVAL));
+
+    *((uint32_t*) val) = SrtVersion(SRT_VERSION_MAJOR, SRT_VERSION_MINOR, SRT_VERSION_PATCH);
+    return 0;
```

Comment:
Why can't you simply return this version by value and have the signature as `uint32_t srt_getversion()`?

Note: you can't have an "error" by getting a version. SRT always has a version and in every possible state.

Reply:
And with a simpler stiff API all possible errors are at compile time.

Reply:
Agreed!